app/services/HistoricalReturnSubmitService.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import config.AppConfig
+import connectors.{CoreVatReturnConnector, RegistrationConnector}
+import logging.Logging
+import models.corrections.CorrectionPayload
+import models.Period
+import uk.gov.hmrc.domain.Vrn
+import utils.ObfuscationUtils.obfuscateVrn
+
+import java.time.{Clock, Instant}
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+trait HistoricalReturnSubmitService
+
+class HistoricalReturnSubmitServiceImpl @Inject()(
+                                               vatReturnService: VatReturnService,
+                                               correctionService: CorrectionService,
+                                               coreVatReturnService: CoreVatReturnService,
+                                               coreVatReturnConnector: CoreVatReturnConnector,
+                                               registrationConnector: RegistrationConnector,
+                                               appConfig: AppConfig,
+                                               clock: Clock
+                                             )
+                                             (implicit ec: ExecutionContext) extends HistoricalReturnSubmitService with Logging {
+
+  val startTransfer: Future[Any] = transfer()
+
+  def transfer(): Future[Any] = {
+    if(appConfig.coreVatReturnsEnabled) {
+      logger.debug("Starting to process all historical returns to send to core")
+
+      vatReturnService.get().flatMap { allReturns =>
```

Comment:
Do we need recovery if either this or correctionService.get() fails? Are there tests for these scenarios

Reply:
It should be caught when using the lift method, here:

https://github.com/hmrc/one-stop-shop-returns/pull/42/files/43578682b4e519ed3d8a6282cc8250c5a73584bd#diff-360b9cf162ce776648226fabcc6cebad01f5a1f9cce4b6da63f4bb12158a6263R83